packages/prisma-client/migrations-cli/umzug.ts

```diff
@@ -0,0 +1,203 @@
+
+const runPrismaCLI = (args: string[]) => {
+  console.info(`Running: prisma ${args.join(" ")}\n`);
+
+  const bin = path.resolve("./node_modules/.bin/prisma");
+  if (fs.existsSync(bin) === false) {
+    console.error(
+      "Could not find Prisma CLI binary. Running `yarn install` might help.\n"
+    );
+    process.exit(1);
+  }
+  try {
+    child_process.execFileSync(bin, args, {
+      stdio: [process.stdin, process.stdout, process.stderr],
+    });
+  } catch (e) {
+    // just exiting, all errors should be already piped into the terminal by Prisma
+    process.exit(e.status);
+  }
+};
+
+const executeSQLFile = async (
+  filepath: string,
+  timeout: number = 1000 * 60
+) => {
+  const source = await fs.promises.readFile(filepath, "utf8");
+
+  // Following advice from here https://github.com/prisma/prisma/issues/2868#issuecomment-650283841
+  // But this is not entirelly safe as it will fail if a query contains ; as a part of a string.
+  const queries = source
+    .split(";")
+    .map((q) => q.trim())
+    .filter((q) => q.length > 0);
+
+  await prisma.$transaction(
+    async (prisma) => {
+      for (const query of queries) {
+        await prisma.$executeRawUnsafe(query);
+      }
+    },
+    { timeout }
+  );
+};
+
+const tmpMemoryStorage = memoryStorage();
+
+export const migrator = new Umzug({
+  migrations: {
+    glob: ["../prisma/migrations/**/*.{ts,sql}", { cwd: __dirname }],
+    resolve(params) {
+      const filePath = params.path;
+
+      // for TypeScript
+      if (filePath == null) {
+        throw new Error("Path is required");
+      }
+
+      const name = path.dirname(filePath).split(path.sep).pop() as string;
+
+      if (path.extname(filePath) === ".sql") {
+        return {
+          name: name,
+          path: filePath,
+          up: async () => {
+            // throw new Error("TEST");
+            // await executeFile(filePath);
+          },
+        };
+      }
+
+      return { ...Umzug.defaultResolver(params), name };
+    },
+  },
+  context: {},
+  storage: {
+    logMigration(params) {
+      // @todo: update in _prisma_migrations as finished
+      return tmpMemoryStorage.logMigration(params);
+    },
+    unlogMigration() {
+      // this is for rollbacks, that we don't currently support
+      throw new Error("unlogMigration not implemented");
+    },
+    executed(meta) {
+      // @todo: read from _prisma_migrations
+      return tmpMemoryStorage.executed(meta);
+    },
+  },
+  create: {
+    template: (filepath) => {
+      const date = new Date();
+      const prefix =
+        date.getFullYear() +
+        [
+          date.getMonth() + 1,
+          date.getDate(),
+          date.getHours(),
+          date.getMinutes(),
+          date.getSeconds(),
+        ]
+          .map((item) => item.toString().padStart(2, "0"))
+          .join("");
+      const name = path.basename(filepath);
+      return [
+        [
+          path.join(
+            __dirname,
```

Comment:
avoid using __dirname, its deprecated in latest node versions

Reply:
Do you know what is the alternative? I googled briefly and couldn't fine a clear answer. If, you don't know, don't worry, I'll look into it more later.

Reply:
https://stackoverflow.com/questions/46745014/alternative-for-dirname-in-node-js-when-using-es6-modules

Reply:
__dirname isn't deprecated https://nodejs.org/api/modules.html#__dirname

You're probably thinking of it being unavailable in the es6-module context. But this code is not in es6-module context.

Reply:
Just not sure `import.meta` will work in the non es6-module, but will try

Reply:
maybe there is a wrapper package that will work everywhere

Reply:
I now use `__dirname` in only one place. Will be easy to replace if we'll need to. Not sure we should spend any more time on this, not really that important.